src/v/pandaproxy/sharded_client_cache.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "config/broker_authn_endpoint.h"
+#include "hashing/jump_consistent_hash.h"
+#include "hashing/xx.h"
+#include "pandaproxy/kafka_client_cache.h"
+#include "pandaproxy/logger.h"
+#include "pandaproxy/types.h"
+
+#include <seastar/core/gate.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/sharded.hh>
+#include <seastar/core/timer.hh>
+
+#include <chrono>
+#include <utility>
+
+namespace pandaproxy {
+
+inline ss::shard_id user_shard(const ss::sstring& name) {
+    auto hash = xxhash_64(name.data(), name.length());
+    return jump_consistent_hash(hash, ss::smp::count);
+}
+
+class sharded_client_cache {
+public:
+    ss::future<> start(
+      ss::smp_service_group sg,
+      YAML::Node const& cfg,
+      config::binding<size_t> max_size,
+      config::binding<std::chrono::milliseconds> keep_alive);
+
+    ss::future<> stop();
+
+    static std::chrono::milliseconds clean_timer();
```

Comment:
unused? also, probably want to rename to clean_timer_period

Reply:
Yes that function was for a unit test. Will remove until I bring unit tests back.